### PR TITLE
support PUK and seed recovery during onboarding

### LINKED ISSUE #12938: Unable to reset Keycard if it was frozen during account recovery (Keycard pairing)
# Bug Report

## Problem 
If Keycard was frozen during account recovery, then there will be an incorrect flow for recovery. 
The only workaround is to reopen the app and recover the account with the mnemonic, and select Keycard as the key storage.

https://user-images.githubusercontent.com/67952253/146794971-c061fae3-3b2a-424d-8ec3-650b2fcecc49.mp4

### Reproduction

1. Create or recover a Keycard multi-account 
2. Re-install Status or open it on another device
3. Accept Status Terms of Use > `Get started` > `Access existing keys` > `Recover with Keycard` > `Begin recovery`
4. Enter invalid 6-digit passcode 3 times
5. On the `Keycard is frozen` screen tap `Reset with mnemonic` (or `with PUK`, no difference)

#### Expected behavior
The `Seed phrase` / `PUK` screen is opened

#### Actual behavior
The user is prompted for a password again

### Additional Information

- Status version: nightly 20/12
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by bitgamma**
Fixes #12938.

Note for QA: please test for regressions in other scenarios where puk/reset recovery is used

### CODE REVIEW

File: src/status_im/multiaccounts/key_storage/core.cljs [hunk 1]
```diff
@@ -109,12 +109,15 @@
                    types/json->clj
                    :error
                    string/blank?
-                   not)]
+                   not)
+        onboarding? (not (or (:multiaccounts/login db) (:multiaccount db)))]
```
Comment (Parveshdhull) [inline src/status_im/multiaccounts/key_storage/core.cljs:113]: Hi @bitgamma,
please can you tell me what `:multiaccounts/login` represents?

If it represents that the user is logging in(not logged yet) then shouldn't logic be?

`onboarding? (or (:multiaccounts/login db) (not (:multiaccount db))))`

Comment (bitgamma) [reply src/status_im/multiaccounts/key_storage/core.cljs:113]: :multiaccounts/login is the account selected for login, while :multiaccount is the logged in multiaccount. If neither is set, I am onboarding, which is what I want to check here

Comment (Parveshdhull) [reply src/status_im/multiaccounts/key_storage/core.cljs:113]: great 👍

**Review by Parveshdhull — APPROVED** [review]

**Comment by qoqobolo** [comment]
@bitgamma thanks for this fix!
The issue itself is fixed, however, I noticed a redirect to a probably unexpected screen during a card resetting

**Steps:**

1. Create or recover a Keycard multi-account
2. Re-install Status or open it on another device
3. Accept Status Terms of Use > Get started > Access existing keys > Recover with Keycard > Begin recovery
4. Enter invalid 6-digit passcode 3 times
5. On the `Keycard is frozen` screen tap `Reset with the mnemonic`
6. Enter seed phrase > Begin setup > Create a 6-digit passcode and confirm it

**Actual result:**
The `Keycard access is reset` pop-up is shown, but the user is redirected to the `Get your keys screen`, so they will have to onboard from the beginning despite the fact that the seed phrase has already been entered and a password has been created

**Expected result:**
The user is redirected to the login screen (unless there are other reasons to keep the current way)


https://user-images.githubusercontent.com/67952253/147470598-82543951-1473-4d5f-9df3-dca345c6b888.mp4

**Comment by bitgamma** [comment]
@qoqobolo that was actually on purpouse. Thing is that the recovery with workflow procedure overwrites all flow information so we don't know what the user was doing. Making the onboarding finishing would be possible, but it would make the already intricate logic even more intricate (short of doing some big refactoring). Since this seems to be a very edge case, bringing the user to the start so they can do whatever they wanted to do seemed the best thing option. What do you think?

**Comment by qoqobolo** [comment]
@bitgamma yeah, I suspected that this was done on purpose, there's no doubt now :) Thanks for the explanation!
PR can be merged (the failed tests are not related to PR).

**Pull request merged**